salt/states/file.py

```diff
@@ -8077,6 +8113,74 @@ def serialize(
     except TypeError:
         pass
 
+    if check_cmd:
+        tmp_filename = salt.utils.files.mkstemp(suffix=tmp_ext, dir=tmp_dir)
+
+        # If file described by name already exists, copy it to the tmp_filename
+        # so we can check for a diff before running check_cmd
+        if __salt__["file.file_exists"](name):
+            try:
+                __salt__["file.copy"](name, tmp_filename)
+            except Exception as exc:  # pylint: disable=broad-except
+                return _error(
+                    ret,
+                    "Unable to copy file {} to {}: {}".format(name, tmp_filename, exc),
+                )
+
+        try:
```

Comment:
It looks like we are re-using a lot of code from the managed function. Can we break this out into a function that both the managed and serialize methods call?